src/dbhelper.py

```diff
@@ -0,0 +1,107 @@
+from PySide import QtSql
+from PySide.QtSql import QSqlQuery
+from passwords import DB_SERVER, DB_PORT, DB_LOGIN, DB_PASSWORD, DB_TABLE
+import logging
+
+# Object to wrap a database connection and provide some vaguely handy utilities.
+_logger = logging.getLogger(__name__)
+
+# Open the database connection. when the module is first touched...
+_db = QtSql.QSqlDatabase("QMYSQL")
+_db.setHostName(DB_SERVER)
+_db.setPort(DB_PORT)
+
+_db.setDatabaseName(DB_TABLE)
+_db.setUserName(DB_LOGIN)
+_db.setPassword(DB_PASSWORD)
+
+_db.setConnectOptions("MYSQL_OPT_RECONNECT=1")
+
+if not _db.open():
+    _logger.error(_db.lastError().text())
+    raise RuntimeError
+
+# A cache of prepared queries, keyed by SQL string. Allows reuse of prepared queries.
+_prepared_queries = {}
```

Comment:
I did, and seemingly this is the optimal approach (it gets funkier when you want to use multiple threads, but we don't, so we can be slightly dumb). This relies on the assumption that these QSqlQuery are semantically equivalent to the ones in QT, and there's not some extra magic hiding somewhere, which seems unlikely?

Prepared statements are cached on the database-side, associated with database connections. Kill the connection and all your prepared statements get wiped out.
Here's our first screwup: We're always opening and closing connections. Given that we only have one thread, we may as well have one connection to rule them all, and have associated with it a prepared statement for everything we ever do.

However, when you do a prepareStatement call it leads to an IPC call to the database (to prepare the statement (or, if already cached, to do nothing)). This is the expense you conveniently dodge by caching the actual objects: one fairly costly IPC transaction per query. (On top of whatever overheads are associated with creating/destroying QSqlQuery objects).
Some guy on StackOverflow advocated this approach, so it *must* be right! :P